libraries/AP_HAL_PX4/CAN.h

```diff
@@ -253,24 +259,33 @@ class PX4CANManager: public AP_HAL::CANManager {
     BusEvent update_event_;
     PX4CAN if0_;
     PX4CAN if1_;
-    uint8_t can_number_;
 
     virtual int16_t select(uavcan::CanSelectMasks& inout_masks,
                            const uavcan::CanFrame* (&pending_tx)[uavcan::MaxCanIfaces], uavcan::MonotonicTime blocking_deadline) override;
 
-    static void initOnce(uint8_t can_number);
+    void initOnce(uint8_t can_number);
 
     bool initialized_;
 
-    static PX4CAN* ifaces[CAN_STM32_NUM_IFACES];
+    PX4CAN* ifaces[CAN_STM32_NUM_IFACES];
+    uint8_t _ifaces_num;
+    uint8_t _ifaces_in_to_out[CAN_STM32_NUM_IFACES];
+    uint8_t _ifaces_out_to_in[CAN_STM32_NUM_IFACES];
+
+    AP_UAVCAN *p_uavcan;
 
 public:
     PX4CANManager() :
-        update_event_(*this), if0_(bxcan::Can[0], update_event_, 0, CAN_STM32_RX_QUEUE_SIZE), if1_(
-            bxcan::Can[1], update_event_, 1, CAN_STM32_RX_QUEUE_SIZE), can_number_(0), initialized_(
+        update_event_(*this), if0_(bxcan::Can[0], nullptr, 0, CAN_STM32_RX_QUEUE_SIZE), if1_(
+            bxcan::Can[1], nullptr, 1, CAN_STM32_RX_QUEUE_SIZE), initialized_(
                 false), p_uavcan(nullptr)
     {
         uavcan::StaticAssert<(CAN_STM32_RX_QUEUE_SIZE <= PX4CAN::MaxRxQueueCapacity)>::check();
+
+        for(uint8_t i = 0; i < CAN_STM32_NUM_IFACES; i++) {
+            _ifaces_out_to_in[i] = 255;
+            _ifaces_in_to_out[i] = 255;
```

Comment:
I didn't understand the comment about one loop. It's more lines but reduces build dependency.